airbyte-integrations/connectors/source-paypal-transaction/source_paypal_transaction/source.py

```diff
@@ -351,9 +382,24 @@ def check_connection(self, logger, config) -> Tuple[bool, any]:
 
         # Try to initiate a stream and validate input date params
         try:
+            # validate input date ranges
             Transactions(authenticator=authenticator, **config).validate_input_dates()
+
+            # validate if Paypal is able to extract data for given start_data
+            start_date = isoparse(config["start_date"])
+            end_date = start_date + timedelta(days=1)
+            stream_slice = {
+                "start_date": start_date.isoformat(),
+                "end_date": end_date.isoformat(),
+            }
+            records = Transactions(authenticator=authenticator, **config).read_records(sync_mode="full", stream_slice=stream_slice)
+            list(records)
+
         except Exception as e:
-            return False, e
+            if "Data for the given start date is not available" in repr(e):
```

Comment:
Probably, will be better to have the class for this instead of hardcode the message?